### ISSUE TITLE
[libevent] Incorrect port version number

### ISSUE THREAD

**Issue by buracchi** [microsoft/vcpkg#32840]
### Description:

The libevent port version 2.1.12+20230128 in vcpkg is incorrectly labeled as 2.1.12, but it is actually a port to version 2.2.1-alpha. This issue was introduced in the PR #29341.

### Expected Behavior:

Any libevent port version 2.1.12 should correspond to the expected version, which is 2.1.12.

### Proposed Solution:

To resolve this issue, the version number for the libevent port 2.1.12+20230128 should be renamed to 2.2.1-alpha#0.

**Comment by JonLiu1993** [comment]
@buracchi, Thanks for posting this issue, 2.2.1-alpha is just a pre-release version, not a stable official version, so according to our policy, it is not recommended to update it to 2.2.1-alpha, the version 2.1.12+20230128 is intended to refer to the current main port The major version is 2.1.12, but because it has not been updated for a long time, the commit of 2023.01.28 was added.

**Comment by dg0yt** [comment]
CC @BillyONeal

**Comment by BillyONeal** [comment]
I agree, `+` should never be used to distinguish anything about versions because they compare equal.

**Comment by dg0yt** [comment]
@BillyONeal The question is if it it is acceptable to use 
~~~json
  "version": "2.2.1-alpha",
  "port-version": 0,
~~~
*now* to indicate that it is really much more an early 2.2.1 than a 2.1.12,  
and to continue bumping `port-version` even when eventually switching to 
~~~json
  "version": "2.2.1",
  "port-version": "123",
~~~
A similar discussion waits for resolution in another issue/PR.

**Comment by PhoebeHui** [comment]
I think this is a port bug, the version should be fixed.

>The question is if it it is acceptable to use

 > "version": "2.2.1-alpha",
>  "port-version": 0,

According to https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/users/versioning.md?plain=1,  it should not be accepted.

**Comment by Neumann-A** [comment]
I feel like the only true resolution of this is to allow:
`version:` and `version-date:` at the same time and give `version:` higher priority than `version-date:`. `port-version: 0,` doesn't actually reflect reality in those cases. 

> According to https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/users/versioning.md?plain=1, it should not be accepted.

Why? It is basically a `version-semver:`
